### PR TITLE
Get/api/articles/article id #14

### PR DESCRIPTION
**Pull request by kam-cheng**
completed ticket #14 by using existing function fetchArticles in models and adding if statements for when article_id exists.

### CODE REVIEW

**Review by Poonam-raj — APPROVED** [review]
Nice code, liked the way you use the same model for all articles and article ID - there is potential for things to get confused as the model isn't obviously for fetching a single article but the code is smart and makes sense.

File: app.js [hunk 1]
```diff
@@ -1,17 +1,43 @@
...
+    res.status(400).send({ msg: "Invalid input of article_id" });
+  else next(err);
+});
+
+//Server error
+app.use((err, req, res, next) => {
+  console.log(err);
+  res.status(500).send("Server Error!");
+});
+
```
Comment (Angaca) [inline app.js:40]: Consider to extract error handling in a separate file

**Review by Angaca — APPROVED** [review]
Good to go but consider my comments!

File: controllers/app.controllers.js [hunk 2]
```diff
@@ -10,6 +14,16 @@ exports.getArticles = async (req, res) => {
   res.status(200).send({ articles });
 };
 
+exports.getArticleById = async (req, res, next) => {
+  try {
+    const { article_id: articleId } = req.params;
+    const article = await fetchArticles(articleId);
+    res.status(200).send({ article: article[0] });
```
[the remark this answers is not in the document]
Comment (kam-cheng) [reply controllers/app.controllers.js]: Thanks - I have moved this over to the models using an additional if statement to return if there is an article id .

**Pull request merged**